streams/src/main/java/org/apache/kafka/streams/processor/internals/SinkNode.java

```diff
@@ -49,8 +49,8 @@ public void addChild(ProcessorNode<?, ?> child) {
     @Override
     public void init(ProcessorContext context) {
         this.context = context;
-        if (this.keySerializer == null) this.keySerializer = (Serializer<K>) context.keySerializer();
-        if (this.valSerializer == null) this.valSerializer = (Serializer<V>) context.valueSerializer();
+        if (this.keySerializer == null) this.keySerializer = (Serializer<K>) context.keySerde().serializer();
+        if (this.valSerializer == null) this.valSerializer = (Serializer<V>) context.keySerde().serializer();
```

Comment:
I really like to see more unit tests for all this code.  The bug above should have been caught by unit tests IMHO, not by code reviews.